kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/StrimziTestClient.java

```diff
@@ -103,20 +103,27 @@ private String waitForConsumer(String namespace, String podName, Duration timeou
         return kubeClient().logsInSpecificNamespace(namespace, podName);
     }
 
-    private List<ConsumerRecord> getConsumerRecords(List<String> logRecords) {
-        return logRecords.stream().map(x -> ConsumerRecord.parseFromJsonString(VALUE_TYPE_REF, x))
-                .filter(Objects::nonNull).map(ConsumerRecord.class::cast).toList();
+    private List<ConsumerRecord> getConsumerRecords(Stream<String> logRecords) {
+        return logRecords.filter(Predicate.not(String::isBlank))
+                .map(x -> ConsumerRecord.parseFromJsonString(VALUE_TYPE_REF, x))
+                .filter(Objects::nonNull)
+                .map(ConsumerRecord.class::cast)
+                .toList();
     }
 
-    private List<String> extractRecordLinesFromLog(String log) {
-        List<String> records = new ArrayList<>();
+    private Stream<String> extractRecordLinesFromLog(String log) {
         String stringToSeek = "Received message:";
 
-        List<String> receivedMessages = Stream.of(log.split("\n")).filter(l -> l.contains(stringToSeek)).toList();
-        for (String receivedMessage : receivedMessages) {
-            records.add(receivedMessage.split(stringToSeek)[1].trim());
-        }
-
-        return records;
+        return Stream.of(log.split("\n"))
+                .filter(l -> l.contains(stringToSeek))
+                .map(line -> {
```

Comment:
the code used to trim, should we do the same still?

Reply:
I don't think the trim actually mattered. I lightly hold the view that if `trim` is important is should be applied by the consumer of the stream, which incidentally this PR does by applying an `isNotBlank` filter...

Reply:
I believe `trim` is important to remove any white space. Currently it is not working in Jenkins, but I'm trying to run them locally as well.